### PR TITLE
updates less and node-sass for NSP related issues

### PR DESCRIPTION
**Pull request by gabrielcsapo**
```
08:25:43 ┌───────────────┬─────────────────────────────────────────────────────────────────┐
08:25:43 │               │ Regular Expression Denial of Service                            │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ Name          │ minimatch                                                       │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ Installed     │ 0.2.14                                                          │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ Vulnerable    │ <=3.0.1                                                         │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ Patched       │ >=3.0.2                                                         │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ Path          │ xoonboardingnodeweb@1.0.0 > kraken-devtools@1.4.0 > minimatch@… │
08:25:43 ├───────────────┼─────────────────────────────────────────────────────────────────┤
08:25:43 │ More Info     │ https://nodesecurity.io/advisories/118                          │
08:25:43 └───────────────┴─────────────────────────────────────────────────────────────────┘
```

> tests

```
LM-NYC-00875272:kraken-devtools gacsapo$ npm test

> kraken-devtools@1.3.0 test /Users/gacsapo/Documents/temp/kraken-devtools
> grunt test

Running "jshint:files" (jshint) task
>> 15 files lint free.

Running "mochaTest:src" (mochaTest) task


  devtools
    ✓ returns a middleware chain using an empty config
    ✓ returns a middleware chain using a non-empty config (58ms)

  plugins:copier
    ✓ copies static files
    ✓ copies nested static files
    ✓ Copies files with no extension
    ✓ Assumes index.html on trailing slash paths
    ✓ Should not barf on a paths with special characters
    ✓ Ignores missing files

  plugins:dustjs
    ✓ compiles to js
    ✓ compiles nested dust to js
    ✓ compiles localized dust to js
SyntaxError: Expected buffer, comment, end of input, partial, raw, reference, section or special but "{" found. At line : 1, column : 12
  at Object.compiler.compile (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/dustjs-linkedin/lib/compiler.js:34:13)
  at /Users/gacsapo/Documents/temp/kraken-devtools/plugins/dustjs.js:56:36
  at /Users/gacsapo/Documents/temp/kraken-devtools/lib/middleware.js:154:9
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)

    ✓ Errors on invalid inputs
    ✓ Errors on missing includes
    ✓ returns a 404 on a non-existent file requested

  plugins:less
    ✓ compiles to css
Error: variable @color is undefined
    ✓ Errors on invalid inputs
Error: 'inc/notfound.less' wasn't found. Tried - css/less/inc/notfound.less,/Users/gacsapo/Documents/temp/kraken-devtools/test/fixtures/public/css/inc/notfound.less,/Users/gacsapo/Documents/temp/kraken-devtools/test/fixtures/public/css/less/inc/notfound.less,inc/notfound.less
    ✓ Errors on missing includes

  plugins:localize
    ✓ should not crash
    ✓ .preHook() regex should process locale for alpha country-codes like HK/zh (HK-zh)
    ✓ .preHook() regex should process locale for alphanumeric country-codes like C2/zh (zh-C2)
    ✓ .preHook() regex should process locale for long country-groups like GROUP/es (es-GROUP)

  plugins:sass
    ✓ compiles to css
    ✓ passes path configuration to compiler
Error: expected ':' after $color in assignment statement
  at options.error (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/node-sass/lib/index.js:272:32)

    ✓ Errors on invalid inputs
Error: File to import not found or unreadable: inc/notfound
Parent style sheet: stdin
  at options.error (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/node-sass/lib/index.js:272:32)

    ✓ Errors on missing includes

  plugins:stylus
    ✓ compiles to css (124ms)
ParseError: /css/stylus/invalid.css:14:19
   10| 
   11| form input
   12| padding: 5px
   13| border: 1px solid
   14| border-radius: 5px
-------------------------^

expected "indent", got "eos"

  at Parser.error (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:259:11)
  at Parser.expect (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:287:12)
  at Parser.block (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:845:12)
  at Parser.selector (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:1555:24)
  at Parser.ident (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:1446:25)
  at Parser.stmt (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:778:26)
  at Parser.statement (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:685:21)
  at Parser.parse (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/parser.js:236:25)
  at Renderer.render (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/renderer.js:80:22)
  at Function.exports.render (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/stylus.js:82:37)
  at styl (/Users/gacsapo/Documents/temp/kraken-devtools/plugins/stylus.js:34:13)
  at /Users/gacsapo/Documents/temp/kraken-devtools/lib/middleware.js:154:9
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)

    ✓ Errors on invalid inputs
Error: /css/stylus/missing.css:1:9
   1| @import 'inc/notfound'
--------------^

failed to locate @import file inc/notfound.styl

  at Evaluator.visitImport (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/evaluator.js:915:21)
  at Evaluator.Visitor.visit (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/index.js:28:40)
  at Evaluator.visit (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/evaluator.js:160:18)
  at Evaluator.visitRoot (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/evaluator.js:707:27)
  at Evaluator.Visitor.visit (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/index.js:28:40)
  at Evaluator.visit (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/evaluator.js:160:18)
  at Evaluator.evaluate (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/visitor/evaluator.js:247:15)
  at Renderer.render (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/renderer.js:86:26)
  at Function.exports.render (/Users/gacsapo/Documents/temp/kraken-devtools/node_modules/stylus/lib/stylus.js:82:37)
  at styl (/Users/gacsapo/Documents/temp/kraken-devtools/plugins/stylus.js:34:13)
  at /Users/gacsapo/Documents/temp/kraken-devtools/lib/middleware.js:154:9
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)

    ✓ Errors on missing includes


  28 passing (402ms)


Done.
```

### CODE REVIEW

File: plugins/less.js [hunk 1]
```diff
@@ -26,25 +26,25 @@ module.exports = function (options) {
     options.ext = options.ext || 'less';
 
     return function (data, args, callback) {
-        var parser = new(lib.Parser)({
+      console.log(args);
```
Comment (jasisk) [inline plugins/less.js]: ⚠️

Comment (gabrielcsapo) [inline plugins/less.js]: what console log? :)

File: plugins/sass.js [hunk 2]
```diff
@@ -28,14 +28,14 @@ module.exports = function (options) {
...
-            success: function success(result) {
-                callback(null, result.css || result); // result.css for ^2
-            },
-            error: function error(err) {
-                err.status = 500; // for ^2
-                callback(err);
-            },
             includePaths: args.paths
+        }, function(err, result) {
+          if (result) {
```
Comment (thefourtheye) [inline plugins/sass.js]: Shouldn't this be `if (err)`?

File: plugins/less.js [hunk 3]
```diff
@@ -26,25 +26,24 @@ module.exports = function (options) {
...
-            parser.parse(data.toString('utf8'), function (err, tree) {
-                if (err) {
-                    callback(err);
-                    return;
-                }
-                callback(null, tree.toCSS());
-            });
+            lib.render(data.toString('utf8'), options)
+              .then(function(tree) {
+                callback(null, tree.css);
```
Comment (thefourtheye) [inline plugins/less.js]: Shouldn't this be `tree.css || tree`?

File: plugins/less.js [hunk 4]
```diff
@@ -26,25 +26,24 @@ module.exports = function (options) {
...
     return function (data, args, callback) {
-        var parser = new(lib.Parser)({
+        var options = {
             paths: args.paths,
             filename: args.context.name,
             dumpLineNumbers: 'comments'
-        });
+        };
 
         try {
```
Comment (thefourtheye) [inline plugins/less.js]: Is the `try...catch` really necessary? I think they don't throw synchronously anymore.

File: plugins/sass.js (hunk 2 above)
Comment (gabrielcsapo) [inline plugins/sass.js]: @thefourtheye the logic would just be inversed? You are suggesting I check for the error and if the error doesn't exist then the result must exist?

File: plugins/less.js (hunk 3 above)
Comment (gabrielcsapo) [inline plugins/less.js]: Tree is an object, `tree.css` Is a string, so I don't know why you would return two different types of values for consistencies sake...

File: plugins/less.js (hunk 4 above)
Comment (gabrielcsapo) [inline plugins/less.js]: The function internally could throw an error and one of the tests even checks for this case to occur.

**Comment by donsalari** [comment]
👍

**Comment by gabrielcsapo** [comment]
@grawk @tlivings thoughts?

**Comment by grawk** [comment]
I'm confused about the scope of the changes here if we are just patching a security threat in `minimatch`. Why the additional changes?

**Comment by grawk** [comment]
`kraken-devtools` is superseded by `construx` and individual plugins. I wonder if we should take this opportunity to deprecate `kraken-devtools` for good and instead work on addressing any security issues in the `construx` modules.

**Comment by gabrielcsapo** [comment]
@grawk works for me. The additional changes are required because those components depend on mini match at that current version, so I needed to bump them. In doing so it caused me to have to integrate those modules via their new interface.

**Comment by gabrielcsapo** [comment]
closing due to lack of need.

**Pull request closed without merging**